packages/service-library/src/servicelib/secure_random.py

```diff
@@ -0,0 +1,26 @@
+import secrets
+
+
+def secure_randint(start: int, end: int) -> int:
+    """Generate a random integer between start (inclusive) and end (exclusive)."""
+    if not isinstance(start, int) or not isinstance(end, int):
+        msg = f"{start=} and {end=} must be integers"
+        raise TypeError(msg)
+    if start >= end:
+        msg = f"{start=} must be less than {end=}"
+        raise ValueError(msg)
+
+    # Calculate the range of possible values
+    num_values = end - start
+
+    # Determine the number of bits required to represent num_values
+    num_bits = num_values.bit_length()
+
+    # Generate random bits until a value within the range is obtained
```

Comment:
Have you really checked `secrets`'s module  functions? I have already used `secrets.randbelow`  so I would not be surprised to find something that fulfill this. 
Having this while loop feels wrong. E.g. the time to compute is not constant  and in principle you do not have guarantee it will hit the condition.

Reply:
yes, this can be achieved.